add tpen-split-screen

git-voo opened the pull request.

thehabes commented on components/split-screen/index.js:
```diff
@@ -0,0 +1,88 @@
+    }
+
+    render() {
+        this.shadowRoot.innerHTML = `
```
This can probably just be in the constructor()

thehabes commented on interfaces/transcription/index.js:
This feels like it is supposed to be in /components, not /interfaces

cubap commented on components/split-screen/index.js:
```diff
@@ -0,0 +1,88 @@
+        const leftPane = this.shadowRoot.querySelector('.left-pane')
+        const rightPane = this.shadowRoot.querySelector('.right-pane')
+        leftPane.style.width = this.leftWidthPercentage + '%'
+        rightPane.style.width = (100 - this.leftWidthPercentage) + '%'
```
This leads to some really fragile behavior.

I think it may be better to set the width of the splitscreen component directly and let the rest of the things resize themselves. The transcription interface can setup an `auto` or `display:flex` scheme to just move when the split pane is set.

cubap commented on components/split-screen/index.js:
```diff
@@ -0,0 +1,88 @@
+            overflow: hidden;
+          }
+          .left-pane {
+            width: ${this.leftWidthPercentage}%;
```
We need to leave `render()` as it's own thing to set up what tool is inside and how it should be operating. The change in width should just be a direct CSS or style change to the existing element or we will break its contents by constantly redrawing them.

cubap commented on interfaces/transcription/index.js:
```diff
@@ -3,7 +3,7 @@ export default class TranscriptionInterface extends HTMLElement {
         this.attachShadow({ mode: "open" })
         this.state = {
-            isSplitscreenActive: true,
+            isSplitscreenActive: false,
```
It might be more sensible to set a splitscreen tool when it is called for. There shouldn't be a way to have "active" split screen with nothing in the pane.

cubap commented on interfaces/transcription/index.js:
```diff
@@ -22,68 +23,95 @@ export default class TranscriptionInterface extends HTMLElement {
+            }
+          </style>
+          <tpen-project-navigation></tpen-project-navigation>
+          <tpen-split-screen>
```
I think the split pane should push aside the transcription component, not contain it. I can be convinced otherwise, but it will be hard.

cubap reviewed (commented):
Good work. This seems to be in the way of some other simpler PRs, unfortunately. It needs attention, but it is close.

cubap reviewed (changes requested):
comments left for changes

git-voo commented:
- Updated TpenSplitScreen to update pane widths via CSS variables (instead of re‑rendering), preserving child content state.

- Modified TranscriptionInterface to use TpenSplitScreen as a separate component that pushes aside transcription content rather than containing it.

- Splitscreen now defaults to off and dispatches a toggle event when closed.
